code/game/objects/machinery/miner.dm

```diff
@@ -0,0 +1,197 @@
+		"<span class='notice'>You fumble around figuring out [src]'s wiring.</span>")
+		var/fumbling_time = 10 SECONDS - 2 SECONDS * user.skills.getRating("engineer")
+		if(!do_after(user, fumbling_time, TRUE, src, BUSY_ICON_UNSKILLED))
+			return FALSE
+	playsound(loc, 'sound/items/wirecutter.ogg', 25, TRUE)
+	user.visible_message("<span class='notice'>[user] starts securing [src]'s wiring.</span>",
+	"<span class='notice'>You start securing [src]'s wiring.</span>")
+	if(!do_after(user, 120, TRUE, src, BUSY_ICON_BUILD))
+		return FALSE
+	if(miner_status != MINER_MEDIUM_DAMAGE)
+		return FALSE
+	playsound(loc, 'sound/items/wirecutter.ogg', 25, TRUE)
+	obj_integrity = 0.66 * max_integrity
+	set_miner_status()
+	user.visible_message("<span class='notice'>[user] secures [src]'s wiring.</span>",
+	"<span class='notice'>You secure [src]'s wiring.</span>")
+	return TRUE
+
+/obj/machinery/miner/wrench_act(mob/living/user, obj/item/I)
+	if(miner_status != MINER_SMALL_DAMAGE)
+
+		return
+	if(user.skills.getRating("engineer") < SKILL_ENGINEER_ENGI)
+		user.visible_message("<span class='notice'>[user] fumbles around figuring out [src]'s tubing and plating.</span>",
+		"<span class='notice'>You fumble around figuring out [src]'s tubing and plating.</span>")
+		var/fumbling_time = 10 SECONDS - 2 SECONDS * user.skills.getRating("engineer")
+		if(!do_after(user, fumbling_time, TRUE, src, BUSY_ICON_UNSKILLED))
+			return FALSE
+	playsound(loc, 'sound/items/ratchet.ogg', 25, TRUE)
+	user.visible_message("<span class='notice'>[user] starts repairing [src]'s tubing and plating.</span>",
+	"<span class='notice'>You start repairing [src]'s tubing and plating.</span>")
+	if(!do_after(user, 150, TRUE, src, BUSY_ICON_BUILD))
+		return FALSE
+	if(miner_status != MINER_SMALL_DAMAGE)
+		return FALSE
+	playsound(loc, 'sound/items/ratchet.ogg', 25, TRUE)
+	obj_integrity = max_integrity
+	set_miner_status()
+	user.visible_message("<span class='notice'>[user] repairs [src]'s tubing and plating.</span>",
+	"<span class='notice'>You repair [src]'s tubing and plating.</span>")
+	start_processing()
+	return TRUE
+
+/obj/machinery/miner/examine(mob/user)
+	. = ..()
+	if(!ishuman(user))
+		return
+	switch(miner_status)
+		if(MINER_DESTROYED)
+			to_chat(user, "<span class='info'>It's heavily damaged, and you can see internal workings.</span>\n<span class='info'>Use a blowtorch, then wirecutters, then a wrench to repair it.</span>")
+		if(MINER_MEDIUM_DAMAGE)
+			to_chat(user, "<span class='info'>It's damaged, and there are broken wires hanging out.</span>\n<span class='info'>Use wirecutters, then wrench to repair it.</span>")
+		if(MINER_SMALL_DAMAGE)
+			to_chat(user, "<span class='info'>It's lightly damaged, and you can see some dents and loose piping.</span>\n<span class='info'>Use a wrench to repair it.</span>")
+		if(MINER_RUNNING)
+			to_chat(user, "<span class='info'>[src]'s internal storage currently has [stored_mineral] sheets stored.</span>")

```

Comment:
Feel free to disregard this suggestion, but maybe checking for engineering skills instead of human?

Reply:
Any human player can fix this, so any human player should also be able to know how to fix it

Reply:
Things like the floodlights work like this: only engineers can at a glance know the state they are in, beyond the visual differences, but anyone can fix them even if they need to fumble more while doing it.
That said, up to you. Resolve it if you don't deem it interesting.